docs/frontend.md

```diff
@@ -1,32 +1,121 @@
+
+## CSS
+
+We [follow the same conventions for writing CSS as used by `govuk-frontend`](https://github.com/alphagov/govuk-frontend/blob/main/docs/contributing/coding-standards/css.md). We lint styles using Stylelint and the rules set out in the `stylint-config-gds` configuration. We use Sass to preprocess CSS files.
+
+Avoid writing CSS where possible – use the standard styles, components and overrides from the [GOV.UK Design System](https://design-system.service.gov.uk) instead.
+
+### Sass
+
+When writing new styles, [use the helpers and mixins provided by `govuk-frontend`](https://frontend.design-system.service.gov.uk/sass-api-reference/).
+
+Stylesheets are organised by the interface they are used by. This ensures that we only serve styles to the users that require them, keeping the overall size of our assets to a minimum.
+
+| Interface                            | Styles                          |
+| ------------------------------------ | ------------------------------- |
+| Shared                               | `app/frontend/styles`           |
+| API docs                             | `app/frontend/styles/api_docs`  |
+| Apply for teacher training           | `app/frontend/styles/candidate` |
+| Manage teacher training applications | `app/frontend/styles/provider`  |
+| Support for apply                    | `app/frontend/styles/support`   |
+
+We keep imports ordered alphabetically (using BEM notation means styles are encapsulated, meaning the order in which styles are imported shouldn’t matter).
+
+## JavaScript
+
+We [follow the same conventions for writing JS as used by `govuk-frontend`](https://github.com/alphagov/govuk-frontend/blob/main/docs/contributing/coding-standards/js.md). We lint styles using StandardJS.
+
+Individual JavaScript modules are bundled together and served only to the interface they are used by. This ensures that we only serve JavaScript to the users that require it, keeping the overall size of our assets to a minimum.
+
+| Interface                            | JavaScript entry point                        |
+| ------------------------------------ | --------------------------------------------- |
+| Shared                               | `app/frontend/packs/application.js`           |
+| API docs                             | `app/frontend/packs/application-api-docs.js`  |
+| Apply for teacher training           | `app/frontend/packs/application-candidate.js` |
+| Manage teacher training applications | `app/frontend/packs/application-provider.js`  |
+| Support for apply                    | `app/frontend/packs/application-support.js`   |
+
+### jQuery
+
+Avoid using jQuery. Browser support for ES5 JavaScript is now widespread enough that a library like jQuery is unnecessary.
+
+### Testing
+
+We test JavaScript using Jest. We also have system specs which test JavaScript using a Chromium headless browser and Capybara.
```

Comment:
Maybe link to Jest docs?

Reply:
Have added links to Jest (StandardJS and Stylelint) websites.